tests/test_transform.py

```diff
@@ -668,13 +668,104 @@ def checkGetJacobian(self, fromName, toName):
                                 err_msg=msg)
 
+    def checkOf(self, fromName, midName, toName):
+        """Test Transform<midName>To<toName>.of(Transform<fromName>To<midName>)
+
+        Parameters
+        ----------
+        fromName : string
+            the prefix of the starting endpoint (e.g., "Point2" for a
+            Point2Endpoint) for the final, concatenated Transform
+        midName : string
+            the prefix for the shared endpoint where two Transforms will be
+            concatenated
+        toName : string
+            the prefix of the ending endpoint for the final, concatenated
+            Transform
+        """
+        transform1Class = getattr(afwGeom,
+                                  "Transform{}To{}".format(fromName, midName))
+        transform2Class = getattr(afwGeom,
+                                  "Transform{}To{}".format(midName, toName))
+        baseMsg = "{}.of({})".format(transform2Class.__name__,
+                                     transform1Class.__name__)
+        for nIn in self.goodNaxes[fromName]:
+            for nMid in self.goodNaxes[midName]:
+                for nOut in self.goodNaxes[toName]:
+                    msg = "{}, nIn={}, nMid={}, nOut={}".format(
+                        baseMsg, nIn, nMid, nOut)
+                    polyMap = makeTwoWayPolyMap(nIn, nMid)
+                    transform1 = transform1Class(polyMap)
+                    polyMap = makeTwoWayPolyMap(nMid, nOut)
+                    transform2 = transform2Class(polyMap)
+                    transform = transform2.of(transform1)
+
+                    fromEndpoint = transform1.getFromEndpoint()
+                    toEndpoint = transform2.getToEndpoint()
+
+                    inPoint = fromEndpoint.pointFromData(makeRawPointData(nIn))
+                    outPointMerged = transform.tranForward(inPoint)
+                    outPointSeparate = transform2.tranForward(
+                        transform1.tranForward(inPoint))
+                    assert_allclose(toEndpoint.dataFromPoint(outPointMerged),
+                                    toEndpoint.dataFromPoint(outPointSeparate),
+                                    err_msg=msg)
+
+                    outPoint = toEndpoint.pointFromData(makeRawPointData(nOut))
+                    inPointMerged = transform.tranInverse(outPoint)
+                    inPointSeparate = transform1.tranInverse(
+                        transform2.tranInverse(outPoint))
+                    assert_allclose(
+                        fromEndpoint.dataFromPoint(inPointMerged),
+                        fromEndpoint.dataFromPoint(inPointSeparate),
+                        err_msg=msg)
+
+        with self.assertRaises(InvalidParameterError):
+            polyMap = makeTwoWayPolyMap(2, 3)
+            transform1 = transform1Class(polyMap)
+            transform2 = transform2Class(polyMap)
+            transform = transform2.of(transform1)
```

Comment:
I suggest rewriting this test so that only the `transform2.of` is expected to fail, and put only that statement into the `with` block. That way you can be sure that a failure occurs in the statement you care about, and that a coding error in the other statements will show up as an unexpected exception.

If I understand correctly you are testing # of axes mismatch, and such a test is only sensible if `midName == "Generic"`. if so I would do something like this:

    # Verify that trying to join the same type of endpoint but mismatched
    # numbers of axes, fails. Note that only GenericEndpoints can have
    # variable numbers of axes.
    if midName == "Generic":
        polyMap = makeTwoWayPolyMap(2, 3)
        transform1 = transform1Class(polyMap)
        transform2 = transform2Class(polyMap)
        with self.assertRaises(InvalidParameterError):
            transform = transform2.of(transform1)